custom/staking/keeper/keeper.go

```diff
@@ -0,0 +1,122 @@
+package keeper
+
+import (
+	abcicometbft "github.com/cometbft/cometbft/abci/types"
+	"github.com/cosmos/cosmos-sdk/codec"
+	stakingkeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	"github.com/cosmos/cosmos-sdk/x/staking/types"
+
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	stakingmiddleware "github.com/notional-labs/composable/v6/x/stakingmiddleware/keeper"
+)
+
+type Keeper struct {
+	stakingkeeper.Keeper
+	cdc               codec.BinaryCodec
+	Stakingmiddleware *stakingmiddleware.Keeper
+	authority         string
+}
+
+func (k Keeper) BlockValidatorUpdates(ctx sdk.Context, height int64) []abcicometbft.ValidatorUpdate {
+	// Calculate validator set changes.
+	//
+	// NOTE: ApplyAndReturnValidatorSetUpdates has to come before
+	// UnbondAllMatureValidatorQueue.
+	// This fixes a bug when the unbonding period is instant (is the case in
+	// some of the tests). The test expected the validator to be completely
+	// unbonded after the Endblocker (go from Bonded -> Unbonding during
+	// ApplyAndReturnValidatorSetUpdates and then Unbonding -> Unbonded during
+	// UnbondAllMatureValidatorQueue).
+	params := k.Stakingmiddleware.GetParams(ctx)
+	shouldExecuteBatch := (height % int64(params.BlocksPerEpoch)) == 0
+	var validatorUpdates []abcicometbft.ValidatorUpdate
+	if shouldExecuteBatch {
+		println("Should Execute ApplyAndReturnValidatorSetUpdates at height : ", height)
```

Comment:
Was this a debug statement? Or should it be `k.Logger(ctx).Info(...)` ?

Reply:
Done